src/background/services/tabState.ts

```diff
@@ -56,4 +58,11 @@ export class TabState {
       state.lastPaymentTimestamp = now
     }
   }
+
+  getSessions(tab: Tabs.Tab) {
+    if (!this.sessions.has(tab)) {
+      this.sessions.set(tab, new Map())
```

Comment:
Even though we're setting in a "get", it helps to keep code clean. Initializing empty Map shouldn't be expensive (in terms of memory or CPU).

Reply:
This is exactly about what I was thinking right now. It looks like whenever we will change the focus from one tab  to another we will perform the `set` (one for `resume` and one for `pause`). The reason I believe we should avoid this, is that people (me included), have a lot of tabs open and we will rely on the service worker to be suspended to actually free up the memory.

Reply:
As the key is `Tab`, memory freeing isn't an issue though.
Empty Map doesn't take much memory.